modules/dnn/src/model.cpp

```diff
@@ -37,7 +37,13 @@ struct Model::Impl
             Mat imInfo = (Mat_<float>(1, 3) << size.height, size.width, 1.6f);
             net.setInput(imInfo, "im_info");
         }
-        net.forward(outs, outNames);
+
+        if (outputNames.empty()) {
+            net.forward(outs, outNames);
+        } else {
+            net.forward(outs, outputNames);
+        }
```

Comment:
Why we need this?

Reply:
I wrote it to keep consistent with the previous `text_deteciton.cpp`, and it gives users a choice to specify the output names.
Maybe it is not necessary, and I have removed it.